cpp/pipelines/add_authority_external_ref.cc

```diff
@@ -130,9 +183,146 @@ void ParseGndWikidataMappingFile(std::string filename,
+                            gnd_data.gnd_id =
+                                gnd_id_temp_string.substr(d_nb_add_str_lenght, (gnd_id_temp_string.length() - d_nb_add_str_lenght));
+
+                            ++total_numbers_of_gnd_id_generated;
+                        }
+                    }
+                }
+                // if id -> about, this means the last of group or this is the closing bracket.
+                // then print the accumulation of last result data and reset all info
+                if (IsThisCloseBracketForId(line_parsed[1])) {
+                    csv_file->write(TextUtil::CSVEscape(gnd_data.gnd_id) + ";");
+                    csv_file->write(TextUtil::CSVEscape(gnd_data.wikidata_personal_entity_id) + ";");
+                    csv_file->write(TextUtil::CSVEscape(gnd_data.wikipedia_personal_address) + "\n");
+                    top_level_number = -1;
+                    second_level_number = -1;
+                    is_start_group = false;
+                    gnd_id_temp_string = "";
+                    gnd_data = {};
+                }
+            }
+
+            if (is_start_group) {
+                // std::cout << top_level_number << " , " << second_level_number << std::endl;
+                if (line_parsed[0][0].get<int>() == top_level_number && line_parsed[0][1].get<int>() == second_level_number) {
+                    if (!line_parsed[1].is_structured()) {
+                        if (line_parsed[1].is_string()) {
+                            if (DoesItMatch(wikipedia_address, line_parsed[1].get<std::string>())) {
+                                gnd_data.wikipedia_personal_address = line_parsed[1].get<std::string>();
+
+                                ++total_number_of_wikipedia;
+                            }
+
+                            // if wikidata
+                            if (DoesItMatch(wikidata_address, line_parsed[1].get<std::string>())) {
+                                wikidata_temp_string = nlohmann::to_string(line_parsed[1]);
+                                gnd_data.wikidata_personal_entity_id = wikidata_temp_string.substr(
+                                    wikidata_address_str_lenght + 1, (wikidata_temp_string.length() - (wikidata_address_str_lenght + 2)));
+
+                                ++total_number_of_wikidata;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        std::cout << "\r"
+                  << "Parsed: " << total_line_parsed << " line(s), "
+                  << " Total GND-ID: " << total_numbers_of_gnd_id_generated << ", Total GND with Wikidata: " << total_number_of_wikidata
+                  << ", Total GND with Wikipedia: " << total_number_of_wikipedia;
+        //   << "GND-ID url: " << tmp_id;
+        std::cout.flush();
+        ++total_line_parsed;
+    }
+
+    auto load_file_end = std::chrono::high_resolution_clock::now();
```

Comment:
same here. load_file_end and duration will never change, so they should be `const`. Also please use constructor notation if possible.